### PR TITLE
move constants into proxy code file where they are used to prevent accidental overwriting

### PR DESCRIPTION
**Pull request by sgtsquiggs**
* moves constants into proxy output file to avoid overwriting when running reinforcer separately per proxy

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -19,12 +19,12 @@ jobs:
         uses: actions/setup-go@v5
         with:
           go-version-file: 'go.mod'
-      -
-        name: Docker Login
-        uses: docker/login-action@v1
-        with:
-          username: ${{ secrets.DOCKERHUB_USERNAME }}
-          password: ${{ secrets.DOCKERHUB_TOKEN }}
+#      -
```
Comment (ethanrgrant) [inline .github/workflows/release.yml:23]: why is this commented but not removed?

File: .goreleaser.yml [hunk 2]
```diff
@@ -37,30 +43,30 @@ snapshot:
...
-  - homepage: https://github.com/clear-street/reinforcer
-    description: "Codegen tool for easy adoption of resiliency patterns in Go"
-    tap:
-      owner: clear-street
-      name: homebrew-reinforcer
-    folder: Formula
-    test: |
-      system "#{bin}/reinforcer --version"
+#dockers:
+#  - goos: linux
```
Comment (ethanrgrant) [inline .goreleaser.yml:48]: same q

File: .github/workflows/release.yml (hunk 1 above)
Comment (sgtsquiggs) [reply .github/workflows/release.yml:23]: Left it in case we wanted to resume publishing docker builds in the future

File: .goreleaser.yml (hunk 2 above)
Comment (sgtsquiggs) [reply .goreleaser.yml:48]: Left it in case we wanted to resume publishing docker builds in the future

File: internal/generator/generator.go [hunk 3]
```diff
@@ -77,17 +77,10 @@ type statement interface {
...
-type fileMeta struct {
-	fileConfig *FileConfig
-	methods    []*method.Method
-}
-
 // Generated contains the code generation out for all the processed types
 type Generated struct {
 	// Common is the golang code that is shared across all generated types
 	Common string
-	// Constants is the golang code that holds constants for compile-time safe references to the proxied methods
```
Comment (tzachshabtay) [inline internal/generator/generator.go:89]: Can you provide more context on the PR? What's the scenario you're fixing? What do you mean by proxy?

File: internal/writer/writer_test.go [hunk 4]
```diff
@@ -19,12 +19,6 @@ func TestWriter_Write(t *testing.T) {
 package mytestpackage
 
 // Common Code Here
-`,
-		Constants: `// Code generated by reinforcer, DO NOT EDIT.
```
Comment (tzachshabtay) [inline internal/writer/writer_test.go:23]: Confused about removing this. Are we not writing this "Code generated by reinforcer" comment in all files anymore?

Comment (sgtsquiggs) [reply internal/writer/writer_test.go:23]: This PR stops generating the Constants file and instead puts each target's constants at the top of the file specific to that target (ie if you generate for `TradeServiceClient` the constants are at the top of `trade_service_client.go`)

I have no idea why they were put into a common file to begin with. It is not deterministic and if you run reinforcer twice in the same directory it OVERWRITES THIS FILE.

File: internal/generator/generator.go (hunk 3 above)
Comment (sgtsquiggs) [reply internal/generator/generator.go:89]: This line is being deleted, Christian referred to the reinforced codegen as a proxy to the original iface/struct/etc.

File: internal/writer/writer_test.go (hunk 4 above)
Comment (sgtsquiggs) [reply internal/writer/writer_test.go:23]: If you look closely you'll see that those constants are being moved to another output file.

**Review by ethanrgrant — APPROVED** [review]

**Review by tzachshabtay — APPROVED** [review]

**Pull request merged**